voucher/views.py

```diff
@@ -170,26 +183,42 @@ def get_eligible_coupon_choices(voucher):
 
     # Check for valid coupon options and return choices
     valid_coupons = [
-        get_valid_coupon_versions(
-            match.product.first(),
-            voucher.user,
-            full_discount=True,
-            company=Company.objects.get(id=settings.VOUCHER_COMPANY_ID),
-        )
-        .filter(coupon__voucher=None)
-        .first()
+        get_valid_voucher_coupons_version(voucher, match.product.first())
         for match in course_matches
     ]
     eligible_choices = [
         (
             json.dumps(
                 (course_matches[i].product.first().id, valid_coupons[i].coupon.id)
             ),
-            course_matches[i].course.title,
+            course_matches[i].title,
         )
         for i in range(len(course_matches))
         if valid_coupons[i] is not None and course_matches[i].product is not None
     ]
     if course_matches and not eligible_choices:
         log.error("Found no valid coupons for matches for voucher %s", voucher.id)
     return eligible_choices
+
+
+def get_valid_voucher_coupons_version(voucher, product):
+    """
+    Return valid coupon versions fo a voucher and product
```

Comment:
`for`

Reply:
Can you move `get_current_voucher`, `get_eligible_coupon_choices`, and `get_valid_voucher_coupons_version` into `utils.py`?